### PR TITLE
new version for implementation of smaller beam pipe

### PR DESCRIPTION
**Pull request by aciarma**
Version to implement the new beam pipe (central part radius=10mm length=18cm).

For the moment I only had to fix the VXD volume definition, as it was depending on the CentralBeamPipe_zmax instead of VertexBarrel_zmax. This caused the innermost layer of the barrel to exit the volume when reducing the central beam pipe half-length.

As for now the VXD is unchanged, but some modifications should be performed in order to benefit of the smaller beam pipe (i.e. have the detector closer to the IP, both radially and longitudinally).

### CODE REVIEW

**Comment by aciarma** [comment]
Changed the thickness of the central chamber to 1.7mm (same thickness of engineered design even if different material). Changed the dimensions of the VXD Barrel in order to fit the small beam pipe. The distance between beam pipe and first layer has been kept the same. The length of the layers has been set to keep the same angular acceptance of the previous version. The number of staves and the offset has been adjusted to fit the new small radius for the first and second Double Layers (outermost DL untouched).

**Comment by gganis** [comment]
Hi Andrea,
A very late reaction on this.
Looks good. Question: why the renaming o2_v02 -> o1_v04 ? For consistency with lcgeo ?

**Comment by vvolkl** [comment]
Yes, indeed, this model is based on `o1` of CLD, and calling it `o2` has already caused confusion

**Comment by andresailer** [comment]
Are these now the same as the lcgeo versions of these detectors with the same name?

File: Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+                <ring r="InnerTracker_Endcap_radius_6"        zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_7x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+5*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x3_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+8*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+13*ITM" zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_12x8_Out" phi0="0"/>
+            </layer>
+
+        </detector>
+
+<!--
+	<detector name="InnerTrackerBarrelSupport" type="TrackerBarrelSupport_o1_v01" id="0"  reflect="true" region="InnerTrackerBarrelRegion">
```
Comment (vvolkl) [inline Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml:485]: Should be 
```suggestion
	<detector name="InnerTrackerBarrelSupport" type="TrackerBarrelSupport_o1_v02" id="0"  reflect="true" region="InnerTrackerBarrelRegion">
```
see https://github.com/iLCSoft/lcgeo/pull/259

File: Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml [hunk 2]
```diff
@@ -0,0 +1,199 @@
+<lccdd>
```
Comment (vvolkl) [inline Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml:1]: This file should go as well to FCCee_o1_v05

File: Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+            <layer id="6">
+                <ring r="InnerTracker_Endcap_radius_6"        zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_7x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+5*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x3_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+8*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+13*ITM" zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_12x8_Out" phi0="0"/>
+            </layer>
+
+        </detector>
+
+<!--
```
[the remark this answers is not in the document]
Comment (aciarma) [reply Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml:484]: Hi Valentin, I dont know I never touched the OT.
It is actually like this also in the o2_v02 repo (which is where I started from).

File: Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml (hunk 2 above)
Comment (aciarma) [reply Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml:1]: I think the beampipe should common for all of the experiments, so it should not go in the DetFCCeeCLD folder but stay in DetFCCeeCommon, right?

**Comment by EmanuelPerez** [comment]
Hi Valentin, Andrea,

I do not know what had been the starting point here.
I remember that some time ago (2.5 yrs or so) I had temporarily removed the supports because I had issues with the corresponding drivers.
Chances are that these were just related to a wrong setup or environment. And I think that these structures should be put in indeed.

Cheers,
E.



On 6 Dec 2022, at 17:01, Valentin Volkl ***@***.******@***.***>> wrote:



@vvolkl commented on this pull request.

________________________________

In Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml<https://github.com/HEP-FCC/FCCDetectors/pull/20#discussion_r1041161807>:

> +                <ring r="InnerTracker_Endcap_radius_5"        zstart="InnerTracker_Endcap_z_5" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_7x5_In"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_5+5*ITM"  zstart="InnerTracker_Endcap_z_5" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x8_In"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_5+13*ITM" zstart="InnerTracker_Endcap_z_5" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_10x4_In"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_5+17*ITM" zstart="InnerTracker_Endcap_z_5" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_11x6_In"  phi0="0"/>
+            </layer>
+
+            <layer id="6">
+                <ring r="InnerTracker_Endcap_radius_6"        zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_7x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+5*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x3_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+8*ITM"  zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_9x5_Out"  phi0="0"/>
+                <ring r="InnerTracker_Endcap_radius_6+13*ITM" zstart="InnerTracker_Endcap_z_6" nmodules="24" dz="3*mm" module="InnerTrackerEndcapModule_12x8_Out" phi0="0"/>
+            </layer>
+
+        </detector>
+
+<!--


@aciarma<https://github.com/aciarma> Why are all the supports removed here?

File: Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml (hunk 1 above)
Comment (aciarma) [reply Detector/DetFCCeeCLD/compact/FCCee_o1_v05/InnerTracker.xml]: Hi Valentin,

modifying this to v02 i get an error when building the detector: "No factory with name Create(TrackerBarrelSupport_o1_v02) for type TrackerBarrelSupport_o1_v02 found." 

what am I missing?

**Comment by aciarma** [comment]
I reverted back to v01 the type of the tracker support as when compiling the factory for the v02 couldn't be found.
Also, as my merge request did not concern the trackers from the start, but only the vertex detector adaptation for the central chamber (which now is R=10mm L=18cm, and it is what it should be used also for the other detectors like IDEA), I hope it can be approved and further modifications be applied in the future.

File: Detector/DetFCCeeCLD/CMakeLists.txt [hunk 4]
```diff
@@ -12,7 +12,7 @@ target_link_libraries(DetFCCeeCLD DD4hep::DDCore)
 set(LIBRARY_OUTPUT_PATH ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 
 add_test(NAME ConstructDetFCCeeCLD
-         COMMAND root -q -b ".github/scripts/load_detector.C(\"Detector/DetFCCeeCLD/compact/FCCee_o2_v02/FCCee_o2_v02.xml\", \"fccee_cld.root\")"
+         COMMAND root -q -b ".github/scripts/load_detector.C(\"Detector/DetFCCeeCLD/compact/FCCee_o2_v02/FCCee_o2_v02.xml\")"
```
Comment (kjvbrt) [inline Detector/DetFCCeeCLD/CMakeLists.txt:15]: Hopefully the removal will not be necessary with https://github.com/HEP-FCC/FCCDetectors/pull/42 .

Comment (aciarma) [reply Detector/DetFCCeeCLD/CMakeLists.txt]: thank you, I updated with the main

**Comment by BrieucF** [comment]
Hi @aciarma ! Is this PR ready to be merged on your side?

**Comment by aciarma** [comment]
> Hi @aciarma ! Is this PR ready to be merged on your side?

Hi @BrieucF, yes for me this is ready

File: Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+        <comment>Part of beampipe made of Beryllium</comment>
+        
+        <detector name="BeBeampipe" type="DD4hep_Beampipe_o1_v01" insideTrackingVolume="true" nocore="true" vis="BeamPipeVis">
+            <parameter crossingangle="CrossingAngle" />
+            <envelope vis="BlueVis">
+                <shape type="Assembly"/>
+            </envelope>
+
+            <section type="Center"               start="0*mm"    end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax"  rMin2="CentralBeamPipe_rmax"    rMax1="CentralBeamPipe_rmax+BeamPipeWidth"  rMax2="CentralBeamPipe_rmax+BeamPipeWidth"                material="Beryllium" name="VertexInnerBe" />
```
Comment (BrieucF) [inline Detector/DetFCCeeCommon/compact/Beampipe_with_notch_noShield.xml:37]: Just out of curiosity, what is the reason for naming the inner start of the beam pipe tube `CentralBeamPipe_rmax` and to add later on the `BeamPipeWidth`? I find it rather confusing... Also why do we need both `rMin1` and `rMin2`? (not asking to change things here in this PR)

**Pull request merged**